applications/MultilevelMonteCarloLaplacianApplication/python_scripts/pure_diffusion_solver.py

```diff
@@ -0,0 +1,241 @@
+
+        
+    def PrepareModelPart(self):
+        if not self.model.HasModelPart(self.settings["model_part_name"].GetString()):
+            self.model.CreateModelPart(self.settings["model_part_name"].GetString())
+            
+        if self._IsPrintingRank():
+            KratosMultiphysics.Logger.PrintInfo("PureDiffusionSolverSolver", "Model reading finished.")
+
+            
+    def GetMinimumBufferSize(self):
+        return 1
+            
+            
+    def Initialize(self):
+        # Creating the solution strategy
+        self.conv_criteria = KratosMultiphysics.DisplacementCriteria(self.settings["relative_tolerance"].GetDouble(),
+                                                                     self.settings["absolute_tolerance"].GetDouble())
+        (self.conv_criteria).SetEchoLevel(self.settings["echo_level"].GetInt())
+        self.time_scheme = KratosMultiphysics.ResidualBasedIncrementalUpdateStaticScheme()
+        self.builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolver(self.linear_solver)
+
+        # self.solver = KratosMultiphysics.ResidualBasedLinearStrategy(self.main_model_part,
+        #                                                              self.time_scheme,
+        #                                                              self.linear_solver,
+        #                                                              self.builder_and_solver,
+        #                                                              self.settings["compute_reactions"].GetBool(),
+        #                                                              self.settings["reform_dofs_at_each_step"].GetBool(),
+        #                                                              self.settings["calculate_norm_dx"].GetBool(),
+        #                                                              self.settings["move_mesh_flag"].GetBool())
+
+        self.solver = KratosMultiphysics.ResidualBasedLinearStrategy(self.main_model_part,
+                                                                     self.time_scheme,
+                                                                     self.linear_solver,
+                                                                     self.builder_and_solver,
+                                                                     False,
+                                                                     False,
+                                                                     False,
+                                                                     False)
+
+
+        (self.solver).SetEchoLevel(self.settings["echo_level"].GetInt())
+        (self.solver).Initialize()
+        (self.solver).Check()
+        # print ("Pure diffusion solver initialization finished")
+
+        
+    def Solve(self):
+        # Solve equations on mesh
+        print("!!!WARNING: YOU SHOULD NOT USE solver.Solve()!!!")
+        (self.solver).Solve()
```

Comment:
see the comment... this should be divided in the InitializeSolutionStep, SolveSolutionStep, etc

Reply:
It is done in the way you are saying, I will delete from the solver this function.